Sources/Purchasing/StoreKit2/StoreKit2TransactionFetcher.swift

```diff
@@ -90,15 +116,144 @@ extension StoreKit.VerificationResult where SignedType == StoreKit.Transaction {
     var verifiedTransaction: StoreKit.Transaction? {
         switch self {
         case let .verified(transaction): return transaction
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(identifier: String(transaction.id), error))
+            return nil
         }
     }
 
-    fileprivate var verifiedStoreTransaction: StoreTransaction? {
+    var verifiedStoreTransaction: StoreTransaction? {
         switch self {
         case let .verified(transaction): return StoreTransaction(sk2Transaction: transaction,
                                                                  jwsRepresentation: self.jwsRepresentation)
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(identifier: String(transaction.id), error))
+            return nil
+        }
+    }
+
+}
+
+// MARK: - Private
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+extension StoreKit.VerificationResult where SignedType == StoreKit.AppTransaction {
+
+    var verifiedAppTransaction: SK2AppTransaction? {
+        switch self {
+        case let .verified(transaction): return .init(appTransaction: transaction)
+        case let .unverified(transaction, error):
+            Logger.warn(
+                Strings.storeKit.sk2_unverified_transaction(identifier: transaction.bundleID, error)
+            )
+            return nil
+        }
+    }
+
+}
+
+extension StoreKit2TransactionFetcher {
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var verifiedTransactions: [StoreTransaction] {
+        get async {
+            return await StoreKit.Transaction.all
+                .compactMap { $0.verifiedStoreTransaction }
+                .extractValues()
+        }
+    }
+
+    /// Returns an an array of  all verified `StoreTransaction`s, making sure `transaction` is contained in it.
+    ///
+    /// When using SKTestSession / SK config files, a newly generated transaction may not appear immediately
+    /// in `Transaction.all`, so we need to retry and introduce a small synthetic delay to make sure it is.s
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private func verifiedTransactions(containing transaction: StoreTransactionType) async -> [StoreTransaction] {
+
+        return await Async.retry {
```

Comment:
This could check a total of 16 times which seems like a lot. Should we increase the delay or reduce the timeout?

Reply:
It's not a timeout it's the maximum number of retries, so it should try max 5 times by default.

Reply:
Oh yeah what am I saying 🤦🏻 this is good then.